core/capabilities/remote/executable/server.go

```diff
@@ -210,10 +212,40 @@ func (r *server) Receive(ctx context.Context, msg *types.MessageBody) {
 	}
 }
 
+func (r *server) logNonDeterministicRequestsWarning(requestIDs map[string]bool, messageID string) {
+	deterministicCapabilityRequests := make([]commoncap.CapabilityRequest, 0, len(requestIDs))
+	for reqID := range requestIDs {
+		req := r.requestIDToRequest[reqID]
+		deterministicCapabilityRequest, err := r.getDeterministicCapabilityRequest(req.message)
+		if err != nil {
+			r.lggr.Errorw("failed to get deterministic capability request", "err", err)
+			continue
+		}
+
+		deterministicCapabilityRequests = append(deterministicCapabilityRequests, deterministicCapabilityRequest)
+	}
+
+	r.lggr.Warnw("received messages with the same id and different capability requests", "messageID", messageID, "uniqueRequestsCount", len(requestIDs), "capabilityRequests", deterministicCapabilityRequests)
```

Comment:
Would this be a problem if each `CapabilityRequest` was a massive size? I'm imagining a 20 node DON all printing the max output.

If so, olutions could be only printing diff, or capping the size.

Reply:
I agree with Justin that we can't do this unfortunately; it could also leak secrets